Conceptpower+Spring/src/test/java/edu/asu/conceptpower/app/db/DBRequestClientTest.java

```diff
@@ -1,53 +1,42 @@
 package edu.asu.conceptpower.app.db;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
-
 import org.junit.Before;
 import org.junit.Test;
+import org.mockito.ArgumentMatchers;
+import org.mockito.InjectMocks;
 import org.mockito.Mock;
 import org.mockito.Mockito;
 import org.mockito.MockitoAnnotations;
 
 import com.db4o.ObjectContainer;
+import com.db4o.query.Predicate;
 
 import edu.asu.conceptpower.core.ReviewRequest;
-import edu.asu.conceptpower.core.ReviewStatus;
 
 public class DBRequestClientTest {
     
-    @Mock
+    @InjectMocks
     private DBRequestClient client;
     
     @Mock
     private ObjectContainer objectContainerClient;
     
-    private ReviewRequest reviewRequest = new ReviewRequest();
     
     @Before
     public void init() {
         MockitoAnnotations.initMocks(this);
-        
-        objectContainerClient = Mockito.mock(ObjectContainer.class);
-        
-        reviewRequest.setConceptId("WID-10126926-N-05-Einstein");
-        reviewRequest.setRequest("Review the Spelling");
-        reviewRequest.setRequester("user2");
-        reviewRequest.setResolver("admin");
-        reviewRequest.setStatus(ReviewStatus.OPENED);
-
-        Mockito.when(client.getReviewRequestForConcept(reviewRequest.getConceptId())).thenReturn(reviewRequest);
     }
+    
+    @Test
+    public void getAllReviewsTest() {
+        client.getAllReviews("WID-10126926-N-05-Einstein");
+        Mockito.verify(objectContainerClient).queryByExample(ArgumentMatchers.any(ReviewRequest.class));
```

Comment:
you want to make sure that queryByExample is called with an example that has the correct id, otherwise your test will succeed even if the queried is is wrong.

Reply:
Since the object instance is created inside the method, we can't check with an equivalent class object. I tried lot of way and ended up with this. Please let me know if we can do it in any other way.

Reply:
https://www.baeldung.com/mockito-argument-matchers section 4 describes your problem I believe.